python/ray/data/preprocessors/encoder.py

```diff
@@ -197,56 +237,70 @@ def __repr__(self):
-        batch = pd.DataFrame(
-            {
-                "A": [["warm"], [], ["hot", "warm", "cold"], ["cold", "cold"]],
-                "B": ["warm", "cold", "hot", "cold"],
-            },
-        )
-        mhe.fit(ray.data.from_pandas(batch))
-        transformed_batch = mhe.transform_batch(batch)
-        expected_batch = pd.DataFrame(
-            {
-                "A": [[0, 0, 1], [0, 0, 0], [1, 1, 1], [2, 0, 0]],
-                "B": [[0, 0, 1], [1, 0, 0], [0, 1, 0], [1, 0, 0]],
-            }
-        )
-        assert transformed_batch.equals(expected_batch)
-
-    Transforming values not included in the fitted dataset or not among
-    the top popular values (see ``max_categories``) will result in all of the encoded
-    column values being 0.
+    """Multi-hot encode categorical data.
+
+    This preprocessor replaces each list of categories with an :math:`m`-length binary
+    list, where :math:`m` is the number of unique categories in the column or the value
+    specified in ``max_categories``. The :math:`i\\text{-th}` element of the binary list
+    is :math:`1` if category :math:`i` is in the input list and :math:`0` otherwise.
+
+    Columns must contain hashable objects or lists of hashable objects.
+    Also, you can't have both types in the same column.
+
+    .. note::
+        The logic is similar to scikit-learn's `MultiLabelBinarizer \
+    <https://scikit-learn.org/stable/modules/generated/sklearn.preprocessing\
+    .MultiLabelBinarizer.html>`_.
+
+    Examples:
+        >>> import pandas as pd
+        >>> import ray
+        >>> from ray.data.preprocessors import MultiHotEncoder
+        >>>
+        >>> df = pd.DataFrame({
+        ...     "name": ["Shaolin Soccer", "Moana", "The Smartest Guys in the Room"],
+        ...     "genre": [
+        ...         ["comedy", "action", "sports"],
+        ...         ["animation", "comedy",  "action"],
+        ...         ["documentary"],
+        ...     ],
+        ... })
+        >>> ds = ray.data.from_pandas(df)  # doctest: +SKIP
+        >>>
+        >>> encoder = MultiHotEncoder(columns=["genre"])
+        >>> encoder.fit_transform(ds).to_pandas()  # doctest: +SKIP
+                                    name            genre
+        0                 Shaolin Soccer  [1, 0, 1, 0, 1]
+        1                          Moana  [1, 1, 1, 0, 0]
+        2  The Smartest Guys in the Room  [0, 0, 0, 1, 0]
+
+        If you specify ``max_categories``, then :class:`MultiHotEncoder`
+        creates features for only the most frequent categories.
+
+        >>> encoder = MultiHotEncoder(columns=["genre"], max_categories={"genre": 3})
+        >>> encoder.fit_transform(ds).to_pandas()  # doctest: +SKIP
+                                    name      genre
+        0                 Shaolin Soccer  [1, 1, 1]
+        1                          Moana  [1, 1, 0]
+        2  The Smartest Guys in the Room  [0, 0, 0]
+        >>> encoder.stats_  # doctest: +SKIP
```

Comment:
Can we document this attribute explicitly?

Reply:
Are we including `Preprocessor` attributes in the public interface? Talked to @matthewdeng and he mentioned it's currently undefined.

I added `stats_` to this example because, without it, there's no way to know which categories are used (`max_categories` doesn't encode infrequent categories).

Reply:
Also, if we're added a attribute to the public interface, we should add `categories_` or `infrequent_categories_`. `stats_` looks weird. See #27357